[hue] Binding broken until you update the new protocol setting to "http" (bridge V1)

I just installed snapshot 3143 containing the PR #11842 merged yesterday.
The binding does no more initialize properly:
```
09:32:18.362 [ERROR] [connection.HueTlsTrustManagerProvider] - An unexpected exception occurred - returning a TrustAllTrustManager: java.net.ConnectException: Connexion refusée (Connection refused)
org.openhab.core.io.net.http.PEMTrustManager$CertificateInstantiationException: java.net.ConnectException: Connexion refusée (Connection refused)
        at org.openhab.core.io.net.http.PEMTrustManager.getPEMCertificateFromServer(PEMTrustManager.java:220) ~[bundleFile:?]
        at org.openhab.core.io.net.http.PEMTrustManager.getInstanceFromServer(PEMTrustManager.java:131) ~[bundleFile:?]
        at org.openhab.core.io.net.http.PEMTrustManager.getInstanceFromServer(PEMTrustManager.java:116) ~[bundleFile:?]
        at org.openhab.binding.hue.internal.connection.HueTlsTrustManagerProvider.getTrustManager(HueTlsTrustManagerProvider.java:63) [bundleFile:?]
        at org.openhab.core.io.net.http.internal.ExtensibleTrustManagerImpl.addTlsTrustManagerProvider(ExtensibleTrustManagerImpl.java:220) [bundleFile:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.invokeMethod(BaseMethod.java:244) [bundleFile:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.access$500(BaseMethod.java:41) [bundleFile:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod$Resolved.invoke(BaseMethod.java:685) [bundleFile:?]
        at org.apache.felix.scr.impl.inject.methods.BaseMethod.invoke(BaseMethod.java:529) [bundleFile:?]
        at org.apache.felix.scr.impl.inject.methods.BindMethod.invoke(BindMethod.java:42) [bundleFile:?]
        at org.apache.felix.scr.impl.manager.DependencyManager.doInvokeBindMethod(DependencyManager.java:2083) [bundleFile:?]
        at org.apache.felix.scr.impl.manager.DependencyManager.invokeBindMethod(DependencyManager.java:2058) [bundleFile:?]
        at org.apache.felix.scr.impl.manager.SingleComponentManager.invokeBindMethod(SingleComponentManager.java:443) [bundleFile:?]
        at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.addedService(DependencyManager.java:333) [bundleFile:?]
        at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.addedService(DependencyManager.java:301) [bundleFile:?]
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1200) [bundleFile:?]
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1121) [bundleFile:?]
        at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.trackAdding(ServiceTracker.java:928) [bundleFile:?]
        at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.track(ServiceTracker.java:864) [bundleFile:?]
        at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:1152) [bundleFile:?]
        at org.apache.felix.scr.impl.BundleComponentActivator$ListenerInfo.serviceChanged(BundleComponentActivator.java:114) [bundleFile:?]
        at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:120) [org.eclipse.osgi-3.17.200.jar:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:956) [org.eclipse.osgi-3.17.200.jar:?]
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:234) [org.eclipse.osgi-3.17.200.jar:?]
        at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:151) [org.eclipse.osgi-3.17.200.jar:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:936) [org.eclipse.osgi-3.17.200.jar:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:873) [org.eclipse.osgi-3.17.200.jar:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.register(ServiceRegistrationImpl.java:141) [org.eclipse.osgi-3.17.200.jar:?]
        at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.registerService(ServiceRegistry.java:261) [org.eclipse.osgi-3.17.200.jar:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:495) [org.eclipse.osgi-3.17.200.jar:?]
        at org.eclipse.osgi.internal.framework.BundleContextImpl.registerService(BundleContextImpl.java:514) [org.eclipse.osgi-3.17.200.jar:?]
        at org.openhab.binding.hue.internal.handler.HueBridgeHandler.initialize(HueBridgeHandler.java:713) [bundleFile:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:154) [bundleFile:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.net.ConnectException: Connexion refusée (Connection refused)
        at java.net.PlainSocketImpl.socketConnect(Native Method) ~[?:?]
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:399) ~[?:?]
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:242) ~[?:?]
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:224) ~[?:?]
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:403) ~[?:?]
        at java.net.Socket.connect(Socket.java:609) ~[?:?]
        at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:289) ~[?:?]
        at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173) ~[?:?]
        at sun.net.NetworkClient.doConnect(NetworkClient.java:182) ~[?:?]
        at sun.net.www.http.HttpClient.openServer(HttpClient.java:474) ~[?:?]
        at sun.net.www.http.HttpClient.openServer(HttpClient.java:569) ~[?:?]
        at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:265) ~[?:?]
        at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:372) ~[?:?]
        at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1187) ~[?:?]
        at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1081) ~[?:?]
        at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177) ~[?:?]
        at sun.net.www.protocol.https.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java:168) ~[?:?]
        at org.openhab.core.io.net.http.PEMTrustManager.getPEMCertificateFromServer(PEMTrustManager.java:209) ~[bundleFile:?]
        ... 45 more
```
The bridge thing remains in status UNKNOWN.

--- a/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/HueTlsTrustManagerProvider.java
+++ b/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/HueTlsTrustManagerProvider.java
@@ -44,6 +44,8 @@ public class HueTlsTrustManagerProvider implements TlsTrustManagerProvider {
 
     private final Logger logger = LoggerFactory.getLogger(HueTlsTrustManagerProvider.class);
 
+    private @Nullable PEMTrustManager trustManager;
+
     public HueTlsTrustManagerProvider(String hostname, boolean useSelfSignedCertificate) {
         this.hostname = hostname;
         this.useSelfSignedCertificate = useSelfSignedCertificate;
@@ -56,20 +58,33 @@ public class HueTlsTrustManagerProvider implements TlsTrustManagerProvider {
 
     @Override
     public X509ExtendedTrustManager getTrustManager() {
+        PEMTrustManager localTrustManager = getPEMTrustManager();
+        if (localTrustManager == null) {
+            logger.error("Cannot get the PEM certificate - returning a TrustAllTrustManager");
+        }
+        return localTrustManager != null ? localTrustManager : TrustAllTrustManager.getInstance();
+    }
+
+    public @Nullable PEMTrustManager getPEMTrustManager() {
+        PEMTrustManager localTrustManager = trustManager;
+        if (localTrustManager != null) {
+            return localTrustManager;
+        }
         try {
             if (useSelfSignedCertificate) {
                 logger.trace("Use self-signed certificate downloaded from Hue Bridge.");
                 // use self-signed certificate downloaded from Hue Bridge
-                return PEMTrustManager.getInstanceFromServer("https://" + getHostName());
+                localTrustManager = PEMTrustManager.getInstanceFromServer("https://" + getHostName());
             } else {
                 logger.trace("Use Signify private CA Certificate for Hue Bridges from resources.");
                 // use Signify private CA Certificate for Hue Bridges from resources
-                return getInstanceFromResource(PEM_FILENAME);
+                localTrustManager = getInstanceFromResource(PEM_FILENAME);
             }
+            this.trustManager = localTrustManager;
         } catch (CertificateException | MalformedURLException e) {
-            logger.error("An unexpected exception occurred - returning a TrustAllTrustManager: {}", e.getMessage(), e);
+            logger.error("An unexpected exception occurred: {}", e.getMessage(), e);
         }
-        return TrustAllTrustManager.getInstance();
+        return localTrustManager;
     }
 
     /**

--- a/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/HueTlsTrustManagerProvider.java
+++ b/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/HueTlsTrustManagerProvider.java
@@ -82,7 +82,7 @@ public class HueTlsTrustManagerProvider implements TlsTrustManagerProvider {
             }
             this.trustManager = localTrustManager;
         } catch (CertificateException | MalformedURLException e) {
-            logger.error("An unexpected exception occurred: {}", e.getMessage(), e);
+            logger.debug("An unexpected exception occurred: {}", e.getMessage(), e);
         }
         return localTrustManager;
     }

--- a/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java
+++ b/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java
@@ -706,20 +706,35 @@ public class HueBridgeHandler extends ConfigStatusBridgeHandler implements HueCl
                     "@text/offline.conf-error-no-ip-address");
         } else {
             if (hueBridge == null) {
-                if (HueBridgeConfig.HTTPS.equals(hueBridgeConfig.protocol)) {
-                    // register trustmanager service
-                    HueTlsTrustManagerProvider tlsTrustManagerProvider = new HueTlsTrustManagerProvider(
-                            ip + ":" + hueBridgeConfig.getPort(), hueBridgeConfig.useSelfSignedCertificate);
-                    serviceRegistration = FrameworkUtil.getBundle(getClass()).getBundleContext()
-                            .registerService(TlsTrustManagerProvider.class.getName(), tlsTrustManagerProvider, null);
-                }
-
                 hueBridge = new HueBridge(httpClient, ip, hueBridgeConfig.getPort(), hueBridgeConfig.protocol,
                         scheduler);
 
                 updateStatus(ThingStatus.UNKNOWN);
+
+                if (HueBridgeConfig.HTTPS.equals(hueBridgeConfig.protocol)) {
+                    scheduler.submit(() -> {
+                        // register trustmanager service
+                        HueTlsTrustManagerProvider tlsTrustManagerProvider = new HueTlsTrustManagerProvider(
+                                ip + ":" + hueBridgeConfig.getPort(), hueBridgeConfig.useSelfSignedCertificate);
+
+                        // Check before registering that the PEM certificate can be downloaded
+                        if (tlsTrustManagerProvider.getPEMTrustManager() == null) {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                                    "@text/offline.conf-error-https-connection");
+                            return;
+                        }
+
+                        serviceRegistration = FrameworkUtil.getBundle(getClass()).getBundleContext().registerService(
+                                TlsTrustManagerProvider.class.getName(), tlsTrustManagerProvider, null);
+
+                        onUpdate();
+                    });
+                } else {
+                    onUpdate();
+                }
+            } else {
+                onUpdate();
             }
-            onUpdate();
         }
     }
 

--- a/bundles/org.openhab.binding.hue/src/main/resources/OH-INF/i18n/hue.properties
+++ b/bundles/org.openhab.binding.hue/src/main/resources/OH-INF/i18n/hue.properties
@@ -148,6 +148,7 @@ config-status.error.missing-ip-address-configuration = No IP address for the Hue
 # thing status descriptions
 
 offline.communication-error = An unexpected exception occurred during execution.
+offline.conf-error-https-connection = HTTPS secure connection failed. Please check your configuration settings (network address, protocol, port, type of certificate) and change configuration to HTTP when using a V1 bridge.
 offline.conf-error-invalid-ssl-certificate = Invalid certificate for secured connection. You might want to enable the "Use Self-Signed Certificate" configuration.
 offline.conf-error-no-ip-address = Cannot connect to Hue Bridge. IP address not available in configuration.
 offline.conf-error-no-username = Cannot connect to Hue Bridge. User name for authentication not available in configuration.

--- a/bundles/org.openhab.binding.hue/src/main/resources/OH-INF/i18n/hue.properties
+++ b/bundles/org.openhab.binding.hue/src/main/resources/OH-INF/i18n/hue.properties
@@ -148,7 +148,7 @@ config-status.error.missing-ip-address-configuration = No IP address for the Hue
 # thing status descriptions
 
 offline.communication-error = An unexpected exception occurred during execution.
-offline.conf-error-https-connection = HTTPS secure connection failed. Please check your configuration settings (network address, protocol, port, type of certificate) and change configuration to HTTP when using a V1 bridge.
+offline.conf-error-https-connection = HTTPS secure connection failed. Please check your configuration settings (network address, protocol, port, type of certificate) and change protocol to http when using a V1 bridge.
 offline.conf-error-invalid-ssl-certificate = Invalid certificate for secured connection. You might want to enable the "Use Self-Signed Certificate" configuration.
 offline.conf-error-no-ip-address = Cannot connect to Hue Bridge. IP address not available in configuration.
 offline.conf-error-no-username = Cannot connect to Hue Bridge. User name for authentication not available in configuration.

